packages/iTwinUI-react/src/core/Slider/Slider.test.tsx

```diff
@@ -458,6 +983,30 @@ it('should show tooltip on thumb hover', () => {
   expect(tippy).not.toBeVisible();
 });
 
+it('should show tooltip on thumb hover (vertical)', () => {
+  const { container } = render(
+    verticalSliderWrapper(
+      <Slider values={defaultSingleValue} orientation='vertical' />,
+    ),
+  );
+  assertBaseElement(container);
+  const thumb = container.querySelector('.iui-slider-thumb') as HTMLDivElement;
+  expect(thumb.classList).not.toContain('iui-active');
+  expect(document.querySelector('.iui-tooltip')).toBeFalsy();
+
+  act(() => {
+    fireEvent.mouseEnter(thumb);
+  });
+  expect(document.querySelector('.iui-tooltip')?.textContent).toBe('50');
+
+  // TODO: What is data-tippy-root? What is the square bracket notation?
+  const tippy = document.querySelector('[data-tippy-root]') as HTMLElement;
+  act(() => {
+    fireEvent.mouseLeave(thumb);
+  });
+  expect(tippy).not.toBeVisible();
+});
```

Comment:
What is tippy here? What is the meaning of the square brackets notation?

Reply:
tippy is a library that we are using for positioning dynamic components like Dropdown menu or Tooltip. In this case it is testing whether tooltip is not visible after focus was moved out from the thumb. And square brackets notation is used to query elements by their attributes so it is querying `<div data-tippy-root>...</div>`. Here is a bit info what you can do with attribute selectors https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors

Reply:
**[DONE]** Ohh, that's a useful selector, thanks